front/src/common/Map/WarpedMap/core/grids.tsx

```diff
@@ -0,0 +1,245 @@
+    border.push(points[0][i]);
+  }
+
+  // Add all intermediary rows:
+  for (let i = 1, l = points.length - 1; i < l; i++) {
+    border.push(points[i][stripsPerSide]);
+    border.unshift(points[i][-stripsPerSide]);
+  }
+
+  // Add last row:
+  for (let i = stripsPerSide; i >= -stripsPerSide; i--) {
+    border.push(lastRow[i]);
+  }
+
+  // Close path:
+  border.push(border[0]);
+
+  return {
+    type: 'polygon',
+    points: border,
+  };
+}
+
+/**
+ * This function takes a path, and returns two isomorphic grids:
+ * - The `regular` grid is vertical grid, with each pair of triangles making a
+ *   perfectly regular triangle (considering the earth is flat, ie lat/lng are
+ *   considered as x/y)
+ * - The `warped` grid follows the path
+ * Using these two grids, it becomes possible to project any point from one grid
+ * to the other.
+ */
+export function getGrids(line: Feature<LineString>, params?: { stripsPerSide?: number }): Grids {
+  const l = line.geometry.coordinates.length;
+  if (l <= 2) throw new Error('line must have at least 3 points');
+
+  const stripsPerSide = params?.stripsPerSide || 2;
+
+  const totalLength = length(line);
+  const step = totalLength / (l - 1);
+  const c = center(line);
+  const flatGrid = featureCollection([]) as GridFeature;
+
+  // Generate flat line:
+  const flatLineStart = destination(c, totalLength / 2, 180);
+  const flatLinePoints: Position[] = [];
+  for (let i = 0; i < l; i++)
+    flatLinePoints.push(destination(flatLineStart, -i * step, 180).geometry.coordinates);
+  const flatLine = lineString(flatLinePoints);
+
+  // Generate flat grid:
+  for (let i = 0; i < l - 1; i++) {
+    const p0 = flatLine.geometry.coordinates[i];
+    const p1 = flatLine.geometry.coordinates[i + 1];
+    for (let direction = -1; direction <= 1; direction += 2) {
+      for (let j = 0; j < stripsPerSide; j++) {
+        const p00 = destination(p0, step * j, direction * 90).geometry.coordinates;
+        const p01 = destination(p0, step * (j + 1), direction * 90).geometry.coordinates;
+        const p10 = destination(p1, step * j, direction * 90).geometry.coordinates;
+        const p11 = destination(p1, step * (j + 1), direction * 90).geometry.coordinates;
+        flatGrid.features.push(
+          polygon([[p00, p10, p01, p00]], {
+            triangleId: `step:${i}/strip:${direction * (j + 1)}/inside`,
+          })
+        );
+        flatGrid.features.push(
+          polygon([[p11, p10, p01, p11]], {
+            triangleId: `step:${i}/strip:${direction * (j + 1)}/outside`,
+          })
+        );
+      }
+    }
+  }
+
+  // Generate "twisted" grid:
+  // 1. Store points for each triangle:
+  const points: PointsGrid = [];
+  for (let i = 0; i < l; i++) {
+    points[i] = {};
+    const p = line.geometry.coordinates[i];
+    const pP = line.geometry.coordinates[i === 0 ? i : i - 1];
+    const pN = line.geometry.coordinates[i === l - 1 ? i : i + 1];
+    const angle = bearing(pP, pN) + 90;
```

Comment:
Can we discuss about this if you have some time ?
```suggestion
    const angle = bearing(pP, pN) + 90;
```